### DISCUSSION TITLE
Magic Link E-mail missing Message-ID header (Custom SMTP)

### DISCUSSION THREAD

**Discussion by Dontosh** [supabase/supabase#30492] [Questions]
I'm using Supabase's Magic Link with Custom STMP setup. Recently, I have heard from people trying out my SvelteKit app that they are not receiving any e-mails. And when I checked my e-mail (which Supabases uses to send the Magic Link) it says that the message could not be delivered because "Messages missing a valid Message-ID header are not accepted". 

Do I have the option to add headers in Supabase for the e-mails that are being sent out, or do I have to use something like SendGrid?

**Comment by ccczy-czy** [comment]
You'd have to configure your SMTP to add messgae-ID in the header. If you're using Supabase's Magic Link with your own Custom STMP setup, then Supabase has no charge of that header setting part. If your SMTP provider can't give you option to add/change the header configuration, then yes you should consider changing to another SMTP provider like SendGrid.
